lib/migration.js

```diff
@@ -295,7 +295,18 @@ function mixinMigration(PostgreSQL) {
     const self = this;
     const modelDef = this.getModelDefinition(model);
     const prop = modelDef.properties[propName];
+
+    // Read column type property
+    const columnMetadata = this.columnMetadata(model, propName);
+    let colType = columnMetadata && columnMetadata.dataType;
+    if (colType) {
+      colType = colType.toUpperCase();
+    }
+
     if (prop.generated) {
+      // For types with prefix BIG should be used BIGSERIAL
+      if (colType && colType.startsWith('BIG'))
```

Comment:
I am worried about a possible confusion this can create for our users, when the framework converts a dataType value `BIGINT` to `BIGSERIAL`.

Why would the developer specify a `BIGINT` data-type when they want to use `BIGSERIAL` instead?

Have you considered to use the `colType` as provided by the user instead of converting all `BIG...` types to `BIGSERIAL`?

Let's discuss.